CASMCMS-9112: Added two SBPS parameters (#5332)
Cray-HPE/docs-csm: operations/iuf/workflows/configuration.md

1. Edit the `compute-and-uan-bootprep.yaml` and `management-bootprep.yaml` files to account for any site deviations from the default values. For example:
    - Comment out the `slurm-site` CFS configuration layer and uncomment the `pbs-site` CFS configuration layer in `compute-and-uan-bootprep.yaml` if PBS is the preferred workload manager

>     - Comment out the SBPS `rootfs_provider` and `rootfs_provider_passthrough` parameters and uncomment the CPS `rootfs_provider` and `rootfs_provider_passthrough` parameters, if CPS is the preferred method to project content.

Reviewer: if DVS with CPS is the preferred method to project content?

Author: Changed.

   There are comments at the top of the `product_vars.yaml` file that describe the variables and related details. The following are a few examples of `site_vars.yaml` changes:
    - Add a `default` section containing a `network_type: "cassini"` entry to designate that Cassini is the desired Slingshot network type to be used when executing CFS configurations later in the workflow
    - Add a `suffix` entry to the `default` section to append a string to the names of CFS configuration, image, and BOS session template artifacts created during the workflow to make them easy to identify

>     - Add a 'system-name' entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this system name as the first part of the domain name. Do not add if not using SBPS.

Reviewer 2: ```suggestion
    - Add a `system-name` entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this system name as the first part of the domain name. Do not add if not using SBPS.
```

>       - See the following page in the CSM documentation:  'operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md'

Reviewer 2: ```suggestion
      - See the following page in the CSM documentation:  `operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md`
```

>       - See the following page in the CSM documentation:  'operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md'

Reviewer 3: We should use a link to this section. Something like..

See the `[Section Title](path/to/section)` procedure for more information.

Author: I fixed these up.

>       - See the procedure [Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS](../../../operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md#boot-set-rootfs_provider_passthrough-parameter) for more information.

Reviewer 2: ```suggestion
      - See the procedure [Create a Session Template to Boot Compute Nodes with SBPS](../../../operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md#boot-set-rootfs_provider_passthrough-parameter) for more information.
```

>         - If the 'docs-csm' RPM is installed on a node, then this page can be found under '/usr/share/doc/csm/'.

Reviewer 2: ```suggestion
        - If the `docs-csm` RPM is installed on a node, then this page can be found under `/usr/share/doc/csm/`.
```

        - Otherwise, it can be found under the appropriate release branch in <https://github.com/Cray-HPE/docs-csm>.

>       - This documentation indicates how to find the system-name.

Reviewer 2: ```suggestion
      - This documentation indicates how to find the `system-name`.
```

>         - If the `docs-csm` RPM is installed on a node, then this page can be found under '/usr/share/doc/csm/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md'. See the `Boot set rootfs_provider_passthrough parameter` section for more details.

Reviewer 2: ```suggestion
        - If the `docs-csm` RPM is installed on a node, then this page can be found under `/usr/share/doc/csm/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md`. See the 
          `Boot set rootfs_provider_passthrough parameter` section for more details.
```

    - Add a 'site-domain' entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this domain name as the second part of the domain name.  Do not add if not using SBPS.

>       - See the following page in the CSM documentation:  'operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md'

Reviewer 3: Same comment as above.

Author: I fixed these up.

>         - If the 'docs-csm' RPM is installed on a node, then this page can be found under '/usr/share/doc/csm/'.

Reviewer 2: ```suggestion
        - If the `docs-csm` RPM is installed on a node, then this page can be found under '/usr/share/doc/csm/'.
```

>     - Add a 'site-domain' entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this domain name as the second part of the domain name.  Do not add if not using SBPS.

Reviewer 2: ```suggestion
    - Add a `site-domain` entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this domain name as the second part of the domain name.  Do not add if not using SBPS.
```

        - Otherwise, it can be found under the appropriate release branch in <https://github.com/Cray-HPE/docs-csm>.

>       - This documentation indicates how to find the site-domain.

Reviewer 2: ```suggestion
      - This documentation indicates how to find the `site-domain`.
```

>       - See the procedure [Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS](../../../operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md#boot-set-rootfs_provider_passthrough-parameter) for more information.

Reviewer 2: ```suggestion
      - See the procedure [Create a Session Template to Boot Compute Nodes with SBPS](../../../operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md#boot-set-rootfs_provider_passthrough-parameter) for more information.
```

>         - If the `docs-csm` RPM is installed on a node, then this page can be found under '/usr/share/doc/csm/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md'. See the `Boot set rootfs_provider_passthrough parameter` section for more details.

Reviewer 2: ```suggestion
        - If the `docs-csm` RPM is installed on a node, then this page can be found under `/usr/share/doc/csm/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md`. See the `Boot set rootfs_provider_passthrough parameter` section for more details.
```

    - Add a `default` section containing a `network_type: "cassini"` entry to designate that Cassini is the desired Slingshot network type to be used when executing CFS configurations later in the workflow
    - Add a `suffix` entry to the `default` section to append a string to the names of CFS configuration, image, and BOS session template artifacts created during the workflow to make them easy to identify
    - Add a `system-name` entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this system name as the first part of the domain name. Do not add if not using SBPS.
      - See the procedure [Create a Session Template to Boot Compute Nodes with SBPS](../../../operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md#boot-set-rootfs_provider_passthrough-parameter) for more information.
        - If the `docs-csm` RPM is installed on a node, then this page can be found under `/usr/share/doc/csm/operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md`. See the 

>           `Boot set rootfs_provider_passthrough parameter` section for more details.

Reviewer 2: ```suggestion
          "Boot set `rootfs_provider_passthrough` parameter" section for more details.
```

   There are comments at the top of the `product_vars.yaml` file that describe the variables and related details. The following are a few examples of `site_vars.yaml` changes:
    - Add a `default` section containing a `network_type: "cassini"` entry to designate that Cassini is the desired Slingshot network type to be used when executing CFS configurations later in the workflow
    - Add a `suffix` entry to the `default` section to append a string to the names of CFS configuration, image, and BOS session template artifacts created during the workflow to make them easy to identify

>     - Add a 'system' entry to the `default` section. The Scalable Boot Projection Service (SBPS) uses this system name as the first part of the domain name. Do not add if not using SBPS.

Reviewer: I think you want it to be `system-name` to match the value below.

Author: Fixed.

       default:
         network_type: "cassini"
         suffix: "-test01"

> 	 system-name: "my-system"

Reviewer: The indentation on these two variables seems incorrect.   Or at least different than the values above

Author: Emacs inserted tabs when I wanted spaces. Fixed.

       ./site_vars.yaml
       ```

> Until Jira CRAYSAT-1900 is resolved, the sat tool will not automatically replace the variables `system-name` and `site-domain` with the values defined in site_vars.yaml in the ${ADMIN_DIR}/bootprep/compute-and-uan-bootprep.yaml file. Therefore, they must

Reviewer 2: ```suggestion
Until Jira CRAYSAT-1900 is resolved, the sat tool will not automatically replace the variables `system-name` and `site-domain` with the values defined in `site_vars.yaml` in the `${ADMIN_DIR}/bootprep/compute-and-uan-bootprep.yaml` file. Therefore, they must
```

> Until Jira CRAYSAT-1900 is resolved, the sat tool will not automatically replace the variables `system-name` and `site-domain` with the values defined in `site_vars.yaml` in the `${ADMIN_DIR}/bootprep/compute-and-uan-bootprep.yaml` file. Therefore, they must

Reviewer 3: Please remove this internal Jira reference. We should avoid this in the public docs.
Maybe start this line with..

SAT will not automatically replace..

Author: Dirty laundry hidden. We do expect the fix to CRAYSAT-1900 imminently, so my plan was never to leave this comment in for long. It was mostly to reassure the installers.

> be replaced manually. Use the same directions listed above for finding the values for these variables to populate the site_vars.yaml file and use them in the ${ADMIN_DIR}/bootprep/compute-and-uan-bootprep.yaml file.

Reviewer 2: ```suggestion
be replaced manually. Use the same directions listed above for finding the values for these variables to populate the site_vars.yaml file and use them in the `${ADMIN_DIR}/bootprep/compute-and-uan-bootprep.yaml` file.
```
